src/main/java/org/fenixedu/academic/domain/dml/DynamicFieldTag.java

```diff
@@ -0,0 +1,66 @@
+package org.fenixedu.academic.domain.dml;
+
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.academic.util.Bundle;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+import org.fenixedu.commons.i18n.LocalizedString;
+
+public class DynamicFieldTag extends DynamicFieldTag_Base {
+
+    private static final String DEFAULT = "DEFAULT";
+
+    private DynamicFieldTag() {
+        super();
+        setRoot(Bennu.getInstance());
+    }
+
+    public static DynamicFieldTag create(String code, LocalizedString name, String domainObjectClassName) {
+        DynamicFieldTag newTag = new DynamicFieldTag();
+
+        newTag.setDomainObjectClassName(domainObjectClassName);
+        newTag.setCode(code);
+        newTag.setName(name);
+
+        return newTag;
+    }
+
+    @Override
+    public void setCode(String code) {
+        Stream<DynamicFieldTag> tagsSet = DynamicFieldTag.findByDomainObjectClassName(getDomainObjectClassName());
+
+        if (tagsSet.anyMatch(tag -> StringUtils.equals(tag.getCode(), code))) {
+            throw new DomainException("error.dynamicFieldTag.duplicated.code");
+        }
+
+        super.setCode(code);
+    }
+
+    public static void setOrCreateDefaultTag(DynamicFieldDescriptor descriptor) {
+        DynamicFieldTag defaultTag = DynamicFieldTag.findByDomainObjectClassName(descriptor.getDomainObjectClassName())
+                .filter(tag -> StringUtils.equals(tag.getCode(), DEFAULT)).findAny().orElse(null);
```

Comment:
we can take advantage of having an optional (_findAny()_) to create the default tag on _"or else"_

Reply:
I am having a reocurring issue where when running the inline "orElse" solution,  despite the optional returned from the FindAny being present, it is still trying to run the code inside the "orElse". 

The current version of the method is working correctly.

Reply:
If you don't want the orElse code to run without it being a orElse situation, then you need to provider a consumer to be run at a later time. Use orElseGet for these scenarios.